### PR TITLE
xds features in gRPC releases

### PR DESCRIPTION
**Pull request by srini100**
Adding the list of xds features in gRPC release.

### CODE REVIEW

File: doc/grpc_xds_features.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+[`envoy.lb.does_not_support_overprovisioning` client
+feature](https://github.com/envoyproxy/envoy/pull/10136) to the xDS
+server to tell the xDS server that it will not perform graceful failover;
+xDS server implementations may use this to decide whether to perform
+graceful failover themselves.
+
+The EDS policy will not support per-endpoint stats; it will report only
+per-locality stats.
+
+An [`lb_endpoint`](https://github.com/envoyproxy/envoy/blob/master/api/envoy/api/v2/endpoint/endpoint_components.proto)
```
Comment (markdroth) [inline doc/grpc_xds_features.md:30]: Suggest having this point to https://github.com/envoyproxy/envoy/blob/12a4bc430eaf440ceb0d11286cfbd4c16b79cdd1/api/envoy/api/v2/endpoint/endpoint_components.proto#L72.

File: doc/grpc_xds_features.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+is ignored if the `health_status` is not HEALTHY or UNKNOWN.
+The optional `load_balancing_weight` is always ignored.
+
+Initially, only `google_default` channel creds will be supported
+to authenticate with the xDS server.
+
+Features | gRFCs  | [C++, Python,<br> Ruby, PHP, C#](https://github.com/grpc/grpc/releases) | [Java](https://github.com/grpc/grpc-java/releases) | [Go](https://github.com/grpc/grpc-go/releases)
+---------|--------|--------------|------|------
+**xDS Infrastructure:**<br>LDS->RDS->CDS->EDS flow,<br>ADS stream,<br>Client-side Load reporting via [LRS](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/service/load_stats/v2/lrs.proto), | [A27](https://github.com/grpc/proposal/blob/master/A27-xds-global-load-balancing.md) | v1.30.0  | v1.30.0 | v1.30.0 |
```
Comment (markdroth) [inline doc/grpc_xds_features.md:39]: The "Client-side load reporting via LRS" part probably belongs in the "Load Balancing" entry, not in the "xDS Infrastructure" entry.

Comment (markdroth) [inline doc/grpc_xds_features.md:39]: Maybe clarify that this xDS infrastructure is only on the gRPC client side, not the server side?

File: doc/grpc_xds_features.md (hunk 1 above)
Comment (srini100) [reply doc/grpc_xds_features.md]: Done.

File: doc/grpc_xds_features.md (hunk 2 above)
Comment (srini100) [reply doc/grpc_xds_features.md]: Done.

Comment (srini100) [reply doc/grpc_xds_features.md]: Done

**Pull request merged**